### PR TITLE
std::get_time and strptime for the truly unfortunate. (Closes #9)

### LINKED ISSUE #9: Windows support ?
See eg [this thread at CCTZ](https://github.com/google/cctz/issues/6).  

This all seems doable, but may need someone with more knowledge of Windows than me.

### PR DESCRIPTION
**Pull request by dcdillon**
This adds a backport of `std::get_time` (in the `std_backports` namespace) and shores up the `strptime()` implementation based on `std::get_time`.

Windows here we come!

### CODE REVIEW

File: src/time_zone_format.cc [hunk 1]
```diff
@@ -31,27 +31,43 @@
 #if !HAS_STRPTIME
 #include <iomanip>
 #include <sstream>
+#include <get_time.h>
```
Comment (eddelbuettel) [inline src/time_zone_format.cc]: Can we make the include conditional on being compiled with MinGW?  Just to minimise side-effects...

File: src/time_zone_format.cc [hunk 2]
```diff
@@ -31,27 +31,43 @@
...
 #endif
 
-namespace cctz {
-namespace detail {
-
-namespace {
+namespace RcppCCTZ {
```
Comment (eddelbuettel) [inline src/time_zone_format.cc]: And maybe keep the existing namespace?

File: src/time_zone_format.cc [hunk 3]
```diff
@@ -500,11 +516,7 @@ const char* ParseSubSeconds(const char* dp,
 // Parses a string into a std::tm using strptime(3).
 const char* ParseTM(const char* dp, const char* fmt, std::tm* tm) {
   if (dp != nullptr) {
-#if defined(__MINGW32__) || defined(__MINGW64__)
-    dp = nullptr;
-#else
-    dp = strptime(dp, fmt, tm);
-#endif    
+    dp = RcppCCTZ::strptime(dp, fmt, tm);
```
Comment (eddelbuettel) [inline src/time_zone_format.cc]: Maybe standard `strptime()` most of the time and our if and when we have no alternative?

File: src/time_zone_format.cc [hunk 4]
```diff
@@ -12,6 +12,8 @@
 //   See the License for the specific language governing permissions and
 //   limitations under the License.
 
+#define __MINGW64__
```
Comment (eddelbuettel) [inline src/time_zone_format.cc]: That one probably needs to go eventually.

File: src/time_zone_format.cc [hunk 5]
```diff
@@ -516,7 +523,7 @@ const char* ParseSubSeconds(const char* dp,
 // Parses a string into a std::tm using strptime(3).
 const char* ParseTM(const char* dp, const char* fmt, std::tm* tm) {
   if (dp != nullptr) {
-    dp = RcppCCTZ::strptime(dp, fmt, tm);
+    dp = cctz::detail::strptime(dp, fmt, tm);
```
Comment (eddelbuettel) [inline src/time_zone_format.cc]: That's a better namespace -- +1

**Comment by dcdillon** [comment]
Lol read the commit comment

**Comment by eddelbuettel** [comment]
Ouch. Damn ipOverPower messing with you?

**Comment by eddelbuettel** [comment]
I am not sure I like the last change set. I don;t think I want to change away from `%F %T` by default. I'd rather condition.

But it looks like you have a build failure to catch anyway?

**Comment by dcdillon** [comment]
Gotcha...no more time tonight.  Will have to address in the morning.  For now, everything should work correctly on every platform.

**Pull request merged**